### DISCUSSION TITLE
ISelect 能否加个 IgnoreColumns()，方便选取指定字段

### DISCUSSION THREAD

**Discussion by ecocom** [dotnetcore/FreeSql#1843] [Ideas]
表中字段比较多时，若要 select 排除一两个字段，现在得把要查询的字段一一写出来：
```c#
// 假设 20 个字段，只排除第二个字段，就得把剩余 19 个一一写出来：
fsql.Select<SomeEntity>().ToList(a => new SomeEntity { Col1 = a.Col1, Col3 = a.Col3, Col4 = a.Col4, Col5 = a.Col5, ...... Col20 = a.Col20 });
```
要是有个 IgnoreColumns，就方便多了：
```c#
fsql.Select<SomeEntity>().IgnoreColumns(a => new { a.Col2 }).ToList();
```

**Comment by densen2014** [comment]
dto啊

**Comment by 2881099** [comment]
.ToList\<Dto\>()

**Reply by ecocom** [reply]
直接得到 List&lt;SomeEntity&gt;，而不是 List&lt;Dto&gt;

**Reply by pjy612** [reply]
好像意思是 
新建一个 SomeDto 里面 是 SomeEntity 缺两个字段。
然后 直接ToList<SomeDto>
之后 可以再用别的工具类转成 SomeEntity ?

**Comment by ecocom** [comment]
不好意思我没有说清楚：
比如某个方法为：
```c#
public virtual IList<SomeEntity> QueryData(......)
{
    ......
}
```
即需要返回 IList&lt;SomeEntity&gt;，无 DTO 设计，而且查询时只排除某一个字段(比如该字段为 BLOB，列表不需要)。

当然可以有多种解决方案，FreeSql 也能解决，比如书写所有字段属性，或 ToList&lt;DTO&gt;() 后再 Convert，或改造 QueryData() 返回类型等等。
我只是说如果有 IgnoreColumns，像这样的需求处理就能方便些，也仅是个小小的建议。

**Reply by pjy612** [reply]
理解理解，因为以前咱好像也提过类似的需求，但后面都还是 迂回 了... （希望少写点类 结果没省掉....

**Reply by pjy612** [reply]
https://github.com/dotnetcore/FreeSql/issues/528
https://github.com/dotnetcore/FreeSql/issues/261

**Reply by densen2014** [reply]
手打的将就看一下

'''
[tablename="class1"]
dto：class1
｛
[colum ( ignore= true)]
new string xxx ｛get；set｝
｝
'''
